### ISSUE TITLE
loss不考虑padsequece的情况？

### ISSUE THREAD

**Issue by nonva** [xuanzebi/BERT-NER#6]
one_hot_labels = tf.one_hot(labels, depth=num_labels, dtype=tf.float32)
per_example_loss = -tf.reduce_sum(one_hot_labels * log_probs, axis=-1)
loss = tf.reduce_sum(per_example_loss)
probabilities = tf.nn.softmax(logits, axis=-1)

log_prob是[batch_size, max_seq, label_num]维度， max_seq有pad，直接reduce_sum全部作为loss？

**Comment by xuanzebi** [comment]
这里可以 在计算loss的时候将padding部分mask掉。

不过当时写的时候因为padding部分idx 为 0，所以在计算loss的时候影响不太大，就没考虑mask.

**Comment by zdgithub** [comment]
> 这里可以 在计算loss的时候将padding部分mask掉。
> 
> 不过当时写的时候因为padding部分idx 为 0，所以在计算loss的时候影响不太大，就没考虑mask.
@xuanzebi 您好，
为什么padding部分的label id=0，在计算loss的时候影响不大？这时one-hot标签向量第0维是1吧

**Comment by xuanzebi** [comment]
想mask就mask掉就好了。其实结果应该都差不太多
loss = tf.reduce_sum(loss * mask) / (tf.reduce_sum(mask) + 1e-8)
